securedrop/encryption.py

```diff
@@ -71,43 +64,23 @@ def __init__(self, gpg_key_dir: Path, journalist_key_fingerprint: str) -> None:
             binary="gpg2", homedir=str(self._gpg_key_dir), options=["--yes", "--trust-model direct"]
         )
 
-        # Ensure that the journalist public key has been previously imported in GPG
-        try:
-            self.get_journalist_public_key()
-        except GpgKeyNotFoundError:
-            raise OSError(
-                f"The journalist public key with fingerprint {journalist_key_fingerprint}"
-                f" has not been imported into GPG."
-            )
-
     @classmethod
     def get_default(cls) -> "EncryptionManager":
         global _default_encryption_mgr
         if _default_encryption_mgr is None:
             config = SecureDropConfig.get_current()
             _default_encryption_mgr = cls(
                 gpg_key_dir=config.GPG_KEY_DIR,
-                journalist_key_fingerprint=config.JOURNALIST_KEY,
+                journalist_pub_key=(config.SECUREDROP_DATA_ROOT / "journalist.pub"),
             )
         return _default_encryption_mgr
 
-    def generate_source_key_pair(self, source_user: "SourceUser") -> None:
-        gen_key_input = self._gpg.gen_key_input(
-            passphrase=source_user.gpg_secret,
-            name_email=source_user.filesystem_id,
-            key_type=self.GPG_KEY_TYPE,
-            key_length=self.GPG_KEY_LENGTH,
-            name_real=self.SOURCE_KEY_NAME,
-            creation_date=self.DEFAULT_KEY_CREATION_DATE.isoformat(),
-            expire_date=self.DEFAULT_KEY_EXPIRATION_DATE,
-        )
-        new_key = self._gpg.gen_key(gen_key_input)
-
-        # Store the newly-created key's fingerprint in Redis for faster lookups
-        self._save_key_fingerprint_to_redis(source_user.filesystem_id, str(new_key))
-
     def delete_source_key_pair(self, source_filesystem_id: str) -> None:
```

Comment:
Nit: I would prefer a docstring like:

```suggestion
    def delete_source_key_pair(self, source_filesystem_id: str) -> None:
        """
        Try to delete the source's key from the filesystem.  If it's not found, either (a) it doesn't exist or (b) the source is Sequoia-based and has its key stored in Source.public_key, which will be deleted when the Source instance itself is deleted.
        """
```

Reply:
Applied